platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/junit/result/TestReportDataCollector.java

```diff
@@ -103,6 +109,13 @@ public void afterTest(TestDescriptor testDescriptor, TestResult result) {
         for (Throwable throwable : result.getExceptions()) {
             methodResult.addFailure(failureMessage(throwable), stackTrace(throwable), exceptionClassName(throwable));
         }
+
+        // TODO: Ugly hack to handle test events without a class
+        // We should make this code handle class-less tests properly
+        if (className == null) {
+            className = "no-class";
+        }
```

Comment:
This is a mistake/oversight in our test descriptors. Let's change https://github.com/gradle/gradle/blob/109e3cec4c515f331081aa7d347d93f8e9d71372/platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/DefaultGroupTestEventReporter.java#L42

to pass name as the className and displayName as the classDisplayName for now.